base_exception/models/base_exception.py

```diff
@@ -63,11 +63,23 @@ def check_exception_type_consistency(self):
                       "domain is missing to match the exception type.")
                 )
 
+    def _get_eval_context(self):
+        """ Prepare the context used when evaluating python code
+            :returns: dict -- evaluation context given to safe_eval
+        """
+        return {
+            'uid': self.env.uid,
+            'user': self.env.user,
+            'env': self.env,
+            'self': self,
+        }
+
```

Comment:
I think it should be only `''ref'": self.env.ref` to limit the risk of evading.

Reply:
Even `ref` returns a recordset which in turn exposes `sudo`.

Reply:
Does that risk also hold for https://github.com/OCA/server-tools/pull/1606?